src/vm/eventpipe.h

```diff
@@ -60,7 +60,11 @@ class EventPipe
     static EventPipeSession *GetSession(EventPipeSessionID id);
 
     // Specifies whether or not the event pipe is enabled.
-    static bool Enabled();
+    static bool Enabled()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return s_tracingInitialized && (s_NumberOfSessions > 0);
```

Comment:
Can you check `s_NumberOfSessions` without taking the lock? It's not `Volatile` and is under the lock in every other usage

Reply:
good call, should make it `Volatile` (cannot use the lock here).